### PR TITLE
Adding functionality to workspace/didChangeConfiguration

### PR DESCRIPTION
**Pull request by victorpigmeo**
- [x] I created an issue to discuss the problem I am trying to solve or an open issue already exists.
- [x] I added a new entry to [CHANGELOG.md](https://github.com/clojure-lsp/clojure-lsp/blob/master/CHANGELOG.md)
- [x] I updated documentation if applicable (`docs` folder)

### CODE REVIEW

**Comment by victorpigmeo** [comment]
Fixes (or at least try to) #1537

File: docs/settings.md [hunk 1]
```diff
@@ -398,3 +398,8 @@ Or to use manual generated stubs:
 ```
 
 clojure-lsp will generate no stubs with that, but analyze that folder and consider it as manual generated stubs.
+
+
+### Changing settings
+
+`clojure-lsp` supports `workspace/didChangeConfiguration` commands to change the server settings while it is running, everything inside the `:settings` map can be changed, but not outside it (the project root for example).
```
Comment (ericdallo) [inline docs/settings.md:405]: maybe worth mentioning that some settings are read only during startup and won't happen nothing if changed after that

File: lib/src/clojure_lsp/handlers.clj [hunk 2]
```diff
@@ -554,3 +554,9 @@
     components (map :old-uri files)
     :will-rename-files
     (f.file-management/will-rename-files files @db*)))
+
+(defn change-server-settings
```
Comment (ericdallo) [inline lib/src/clojure_lsp/handlers.clj:558]: WDYT about naming as `did-change-configuration` to follow the name standard?

File: docs/settings.md (hunk 1 above)
Comment (victorpigmeo) [reply docs/settings.md:405]: nice, do we have a list of these settings?

File: lib/test/clojure_lsp/feature/change_server_settings_test.clj [hunk 3]
```diff
@@ -0,0 +1,120 @@
+(ns clojure-lsp.feature.change-server-settings-test
```
[the remark this answers is not in the document]
Comment (victorpigmeo) [reply lib/test/clojure_lsp/feature/change_server_settings_test.clj:1]: hmm, makes sence to me, I'll move it

**Review by ericdallo — APPROVED** [review]

**Comment by ericdallo** [comment]
@victorpigmeo it seems there is still a failing test because of this PR

**Comment by victorpigmeo** [comment]
yeah, it is failing on windows, sorry I got no time to deal with it yet, but I'll try to fix this evening

**Review by ericdallo — APPROVED** [review]
Thank you!

**Pull request merged**